libcodechecker/analyze/log_parser.py

```diff
@@ -27,13 +28,84 @@
      '^-std=.*'])
 
 
-def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
-    """
-    Returns a list of default includes of the given compiler.
-    """
+def get_compiler_err(cmd):
+    '''
+    Returns the stderr of a compiler invocation as string.
+    '''
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        _, err = proc.communicate("")
+        return err
+    except OSError as oerr:
+        LOG.error("Error during process execution: " + cmd + '\n' +
+                  oerr.strerror + "\n")
+
+
+def parse_compiler_includes(lines):
+    '''
+    Parse the compiler include paths from a string
+    '''
     start_mark = "#include <...> search starts here:"
     end_mark = "End of search list."
 
+    include_paths = []
+
+    do_append = False
+    for line in lines.splitlines(True):
+        line = line.strip()
+        if line.startswith(end_mark):
+            do_append = False
+        if do_append:
+            # On OSX there are framework includes,
+            # where we need to strip the "(framework directory)" string.
+            # For instance:
+            # /System/Library/Frameworks (framework directory)
+            fpos = line.find("(framework directory)")
+            if fpos == -1:
+                include_paths.append("-isystem " + line)
+            else:
+                include_paths.append("-isystem " + line[0:fpos-1])
+
+        if line.startswith(start_mark):
+            do_append = True
+
+    return include_paths
+
+
+def parse_compiler_target(lines):
+    """
+    Parse the compiler target from a string.
+
+    """
+    target_label = "Target:"
+    target = ""
+
+    for line in lines.splitlines(True):
+        line = line.strip().split()
+        if line[0] == target_label:
+            target = line[1]
+
+    return target
+
+
+def dump_compiler_info(output_path, filename, data):
```

Comment:
If the file already exists, we just append to it.
Is it ensured that multiple concurrent checks (to different result directories) won't interfere?
When will we clean up this file? If we don't do that, this file will just grow with subsequent checks to the same result directory.